pallets/fusion/src/lib.rs

```diff
@@ -0,0 +1,3085 @@
+		CannotSetCompoudingWithLessThanMinimum,
+		/// The state cannot be set to open if the pool is not nominating
+		PoolIsNotNominating,
+		/// The controller of the slash destination can only be set with the correct extrinsic
+		CannotSetControllerForSlashDestination,
+		/// A user tried to claim but the account is empty, can try again later
+		NotEnoughClaimableBalanceInPool,
+		/// The maximum TVL was reached
+		MaxTVLReached,
+		/// No valid validators was provided in the targets
+		NoValidValidators,
+		/// Era duration was not recorded properly so we cannot retry
+		EraDurationNotFound,
+		/// Pool has leftover funds, but we did not specify where it should go
+		NoLeftoverDestinationProvided,
+		/// The limit in the pool pending slashes have been reached
+		PendingSlashLimitReached,
+		/// Slash not found in pool
+		SlashNotFound,
+		/// The user does not have a membership in the AVAIL pool.
+		NoAvailMembership,
+		/// The pool does not have boost configured.
+		PoolHasNoBoost,
+		/// The user does not have enough AVAIL to allocate to the boosted pools.
+		NotEnoughAvailForBoost,
+		/// The TC canno't set a controller address for a user, it can only remove (to clean)
+		RootCanOnlyRemoveController,
+		/// TODO Temp, we'll see when bridge com is done
+		CannotDepositAvailCurrency,
+	}
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		fn integrity_test() {
+			// Checks that Fusion bonding duration is greater than slash defer duration
+			assert!(
+				T::SlashDeferDuration::get() < T::BondingDuration::get() || T::BondingDuration::get() == 0,
+				"As per documentation, slash defer duration ({}) should be less than bonding duration ({}).",
+				T::SlashDeferDuration::get(),
+				T::BondingDuration::get(),
+			)
+		}
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// TODO - Dummy extrinsic to add currency without bridge, to be removed
+		#[pallet::call_index(99)]
+		#[pallet::weight(T::WeightInfo::create_currency())]
+		pub fn deposit_currency_dummy(
+			origin: OriginFor<T>,
+			fusion_address: FusionAddress,
+			currency_id: CurrencyId,
+			amount: FusionCurrencyBalance,
+		) -> DispatchResult {
+			ensure_signed(origin)?;
+			Self::do_deposit_currency(fusion_address, currency_id, amount)?;
+			Ok(())
+		}
+
+		/// Creates a new currency
+		#[pallet::call_index(0)]
+		#[pallet::weight(T::WeightInfo::create_currency())]
+		pub fn create_currency(
+			origin: OriginFor<T>,
+			currency_id: CurrencyId,
+			name: BoundedVec<u8, T::MaxCurrencyNameLength>,
+			nb_decimals: u8,
+			max_amount: FusionCurrencyBalance,
+			min_amount: FusionCurrencyBalance,
+			initial_conversion_rate: BalanceOf<T>,
+		) -> DispatchResult {
+			ensure_root(origin)?;
+
+			ensure!(
+				!Currencies::<T>::contains_key(currency_id),
+				Error::<T>::CurrencyAlreadyExists
+			);
+
+			ensure!(name.len() > 0, Error::<T>::InvalidName);
+			ensure!(nb_decimals > 0, Error::<T>::InvalidNumberOfDecimals);
```

Comment:
Is this a requirement? Is there something wrong with having currencies without decimal places?

Reply:
There is not currency without decimals place in Crypto. I would have liked to say something like, minimum amount of decimals is 4 or 8, in the mean time, greater than zero is fine